### PR TITLE
Color command validation

### PR DESCRIPTION
**Pull request by IAmBitfrost**
validate that provided hex color code is at least somewhat readable on all discord themes

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "rusty-bot",
-  "version": "4.2.2",
+  "version": "4.2.3",
```
Comment (Mr1upMachine) [inline package.json:3]: `package-lock.json` should also change with the version increment. Run `npm i`

File: src/utilities/hex-color-helper.ts [hunk 2]
```diff
@@ -26,3 +28,55 @@ export function formatHexColor(color: string): HexColor {
...
+  ];
+}
+
+function contrast(rgb1: RGB, rgb2: RGB): number {
+  var lum1 = luminance(rgb1[0], rgb1[1], rgb1[2]);
+  var lum2 = luminance(rgb2[0], rgb2[1], rgb2[2]);
+  var brightest = Math.max(lum1, lum2);
+  var darkest = Math.min(lum1, lum2);
+
+  console.log((brightest + 0.05) / (darkest + 0.05));
```
Comment (Mr1upMachine) [inline src/utilities/hex-color-helper.ts:72]: Do we need to log the color here?

File: src/utilities/hex-color-helper.ts [hunk 3]
```diff
@@ -26,3 +28,55 @@ export function formatHexColor(color: string): HexColor {
...
+  return (brightest + 0.05) / (darkest + 0.05);
+}
+
+function luminance(r: number, g: number, b: number): number {
+  var a = [r, g, b].map(function (v) {
```
Comment (Mr1upMachine) [inline src/utilities/hex-color-helper.ts:77]: I know you copied these methods but is it possible we can swap with `let` or `const`? When I implement some of the eslint rules I know one will be disallowing `var`

**Review by Mr1upMachine — CHANGES_REQUESTED** [review]

File: src/utilities/hex-color-helper.ts (hunk 3 above)
Comment (TheNickmaster21) [reply src/utilities/hex-color-helper.ts:77]: Definitely agree with now allowing var. Also, I would recommend using the () => { syntax as well.

File: src/utilities/hex-color-helper.ts [hunk 4]
```diff
@@ -26,3 +28,55 @@ export function formatHexColor(color: string): HexColor {
...
+
+  return [
+    parseInt(hexComponents[0], 16),
+    parseInt(hexComponents[1], 16),
+    parseInt(hexComponents[2], 16)
+  ];
+}
+
+function contrast(rgb1: RGB, rgb2: RGB): number {
+  var lum1 = luminance(rgb1[0], rgb1[1], rgb1[2]);
```
Comment (TheNickmaster21) [inline src/utilities/hex-color-helper.ts:67]: Let's replace these with const

**Review by TheNickmaster21 — CHANGES_REQUESTED** [review]

**Review by Mr1upMachine — APPROVED** [review]

**Review by TheNickmaster21 — APPROVED** [review]

**Pull request merged**